client/src/stores/collectionElementsStore.ts

```diff
@@ -9,24 +9,32 @@ export const useCollectionElementsStore = defineStore("collectionElementsStore",
     const loadingCollectionElements = ref<{ [key: string]: boolean }>({});
     const storedCollectionElements = ref<{ [key: string]: DCESummary[] }>({});
 
+    /**
+     * Returns a key that can be used to store or retrieve the elements of a collection in the store.
+     */
+    function getCollectionKey(collection: HDCASummary) {
+        return `${collection.id}-${collection.collection_id}`;
```

Comment:
So this seems pretty weird to me. Are you storing multiple types of collections (HDCA / DatasetCollection / DatasetCollectionElement) in the store ?

These are all distinct, and in practice maybe this works, but if you're using this function on a DatasetCollectionElement you instead of a HDCA you may still get the wrong item.

Reply:
Yes, I agree, and most likely I am still mixing concepts :disappointed: 
My aim was to be able to call `/api/dataset_collections/{hdca_id}/contents/{collection_id}` from any `HDCA` or `DatasetCollection` (`DatasetCollectionElements` are skipped when using `onViewSubCollection`, only their name is used along with the `object`). But I'm not even sure this is how it is supposed to query the elements of subcollections.

Reply:
In the latest changes, I think I've addressed this by consistently using the `DatasetCollection` ID as the key for the elements regardless of the source collection we are querying (top-level HDCA or sub-collection)